### PR TITLE
test(mme): refactoring function is_num_sessions_valid() inside spgw_test_util

### PR DESCRIPTION
**Pull request by thhung**
Signed-off-by: Hung NGUYEN <nguyen.thanh.hung.91@gmail.com>

## Summary

- Refactoring the function is_num_sessions_valid inside the unittest util of spgw into 2 functions:

1. is_num_ue_contexts_valid(): validating the number of ue contexts for a given ue
2. is_num_teids_valid(): validating the number of teid of a given ue

## Test Plan

- make test_oai

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/c/core/oai/test/spgw_task/spgw_test_util.cpp [hunk 1]
```diff
@@ -62,6 +62,29 @@ bool is_num_sessions_valid(
   return true;
 }
 
+bool is_num_ue_contexts_valid(uint64_t imsi64, int expected_num_ue_contexts) {
```
Comment (ssanadhya) [inline lte/gateway/c/core/oai/test/spgw_task/spgw_test_util.cpp]: No need for Imsi here

```suggestion
bool is_num_ue_contexts_valid(int expected_num_ue_contexts) {
```

**Review by ssanadhya — COMMENTED** [review]
@thhung , thanks for adding these functions! Please replace usage of `is_num_sessions_valid` with these new ones in other tests too.

File: lte/gateway/c/core/oai/test/spgw_task/test_spgw_procedures_with_injected_state.cpp [hunk 2]
```diff
@@ -195,6 +195,9 @@ TEST_F(SPGWAppInjectedStateProcedureTest, TestDeleteSessionSuccess) {
 
   ASSERT_TRUE(eps_bearer_ctxt_p->paa.ipv4_address.s_addr == test_ue_ip);
 
+  ASSERT_TRUE(is_num_ue_contexts_valid(test_imsi64, name_of_ue_samples.size()));
+  ASSERT_TRUE(is_num_teids_valid(test_imsi64, 1));
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/test/spgw_task/test_spgw_procedures_with_injected_state.cpp]: Can we rename as is_num_cp_teids_valid to indicate that we are validating control plane teids

File: lte/gateway/c/core/oai/test/spgw_task/spgw_test_util.cpp (hunk 1 above)
Comment (thhung) [reply lte/gateway/c/core/oai/test/spgw_task/spgw_test_util.cpp]: Removed

File: lte/gateway/c/core/oai/test/spgw_task/test_spgw_procedures_with_injected_state.cpp (hunk 2 above)
Comment (thhung) [reply lte/gateway/c/core/oai/test/spgw_task/test_spgw_procedures_with_injected_state.cpp]: Thanks for your review. I will rename this function to clarify it.

File: lte/gateway/c/core/oai/test/spgw_task/spgw_test_util.h [hunk 3]
```diff
@@ -44,6 +44,10 @@ namespace lte {
 bool is_num_sessions_valid(uint64_t imsi64, int expected_num_ue_contexts,
```
Comment (ssanadhya) [inline lte/gateway/c/core/oai/test/spgw_task/spgw_test_util.h]: Please remove this function as it is replaced by the new functions in this PR

**Review by ssanadhya — CHANGES_REQUESTED** [review]

**Review by ssanadhya — APPROVED** [review]
LGTM

**Pull request merged**